src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/miner/ActiveMiner.java

```diff
@@ -0,0 +1,362 @@
+
+        queue.thenRun(6, () -> setPistonState(pistons[0], true));
+        queue.thenRun(9, () -> setPistonState(pistons[0], false));
+
+        queue.thenRun(4, () -> setPistonState(pistons[1], true));
+        queue.thenRun(7, () -> setPistonState(pistons[1], false));
+
+        queue.thenRun(3, () -> setPistonState(pistons[0], true));
+        queue.thenRun(5, () -> setPistonState(pistons[0], false));
+
+        queue.thenRun(2, () -> setPistonState(pistons[1], true));
+        queue.thenRun(4, () -> setPistonState(pistons[1], false));
+
+        queue.thenRun(1, () -> setPistonState(pistons[0], true));
+        queue.thenRun(3, () -> setPistonState(pistons[0], false));
+
+        queue.thenRun(1, () -> setPistonState(pistons[1], true));
+        queue.thenRun(2, () -> setPistonState(pistons[1], false));
+
+        queue.thenRun(1, this);
+        queue.execute(SlimefunPlugin.instance);
+    }
+
+    @Override
+    public void run() {
+        if (!running) {
+            // Don't continue if the machine has stopped
+            return;
+        }
+
+        try {
+            TaskQueue queue = new TaskQueue();
+
+            queue.thenRun(1, () -> setPistonState(pistons[0], true));
+            queue.thenRun(3, () -> setPistonState(pistons[0], false));
+
+            queue.thenRun(1, () -> setPistonState(pistons[1], true));
+            queue.thenRun(3, () -> setPistonState(pistons[1], false));
+
+            queue.thenRun(() -> {
+                Block furnace = chest.getRelative(BlockFace.DOWN);
+                furnace.getWorld().playEffect(furnace.getLocation(), Effect.STEP_SOUND, Material.STONE);
+
+                for (int y = height; y > 0; y--) {
+                    Block b = start.getWorld().getBlockAt(x, y, z);
+
+                    if (!SlimefunPlugin.getProtectionManager().hasPermission(Bukkit.getOfflinePlayer(owner), b, ProtectableAction.BREAK_BLOCK)) {
+                        stop("machines.INDUSTRIAL_MINER.no-permission");
+                        return;
+                    }
+
+                    if (miner.canMine(b.getType()) && push(miner.getOutcome(b.getType()))) {
+                        furnace.getWorld().playEffect(furnace.getLocation(), Effect.STEP_SOUND, b.getType());
+                        furnace.getWorld().playSound(furnace.getLocation(), Sound.ENTITY_ARROW_HIT_PLAYER, 0.2F, 1F);
+
+                        b.setType(Material.AIR);
+                        fuel--;
+                        ores++;
+
+                        // Repeat the same column when we hit an ore.
+                        Slimefun.runSync(this, 4);
+                        return;
+                    }
+                }
+
+                nextColumn();
+            });
+
+            queue.execute(SlimefunPlugin.instance);
+        }
+        catch (Exception e) {
+            Slimefun.getLogger().log(Level.SEVERE, "An Error occured while running an Industrial Miner", e);
```

Comment:
We should post location info like in cargo. Better yet, have a util method in ErrorCatcher to handle it for us.

Think we should always post coords if possible, server owners may need them depending on the issue.